test/hotspot/jtreg/compiler/lib/compile_framework/CompileFramework.java

```diff
@@ -0,0 +1,303 @@
+        println("File: " + path.toString());
+
+        // Ensure directory of the file exists.
+        Path dir = path.getParent();
+        try {
+            Files.createDirectories(dir);
+        } catch (Exception e) {
+            throw new CompileFrameworkException("Could not create directory: " + dir.toString(), e);
+        }
+
+        // Write to file.
+        try (BufferedWriter writer = Files.newBufferedWriter(path)) {
+            writer.write(code);
+        } catch (Exception e) {
+            throw new CompileFrameworkException("Could not write file: " + path.toString(), e);
+        }
+    }
+
+    private static void executeCompileCommand(List<String> command) {
+        println("Compile command: " + String.join(" ", command));
+
+        ProcessBuilder builder = new ProcessBuilder(command);
+        builder.redirectErrorStream(true);
+
+        String output;
+        int exitCode;
+        try {
+            Process process = builder.start();
+            boolean exited = process.waitFor(COMPILE_TIMEOUT, TimeUnit.SECONDS);
+            if (!exited) {
+                process.destroyForcibly();
+                System.out.println("Timeout: compile command: " + String.join(" ", command));
+                throw new InternalCompileFrameworkException("Process timeout: compilation took too long.");
+            }
+            output = new String(process.getInputStream().readAllBytes(), StandardCharsets.UTF_8);
+            exitCode = process.exitValue();
+        } catch (IOException e) {
+            throw new InternalCompileFrameworkException("IOException during compilation", e);
+        } catch (InterruptedException e) {
+            throw new CompileFrameworkException("InterruptedException during compilation", e);
+        }
+
+        if (exitCode != 0 || !output.equals("")) {
+            System.out.println("Compilation failed.");
+            System.out.println("Exit code: " + exitCode);
+            System.out.println("Output: '" + output + "'");
+            throw new CompileFrameworkException("Compilation failed.");
+        }
+    }
+
+    private void setUpClassLoader() {
+        ClassLoader sysLoader = ClassLoader.getSystemClassLoader();
+
+        try {
+            // Classpath for all included classes (e.g. IR Framework).
+            // Get all class paths, convert to urls.
+            List<URL> urls = new ArrayList<URL>();
+            for (String path : getClassPaths()) {
+                urls.add(new File(path).toURI().toURL());
+            }
+            // And add in the compiled classes from this instance of CompileFramework.
+            urls.add(new File(classesDir.toString()).toURI().toURL());
+            classLoader = URLClassLoader.newInstance(urls.toArray(URL[]::new), sysLoader);
+        } catch (IOException e) {
+            throw new CompileFrameworkException("IOException while creating ClassLoader", e);
+        }
+    }
+
+    public Class getClass(String name) {
```

Comment:
You should not use `Class` without type parameter. Suggestion:
```suggestion
    public Class<?> getClass(String name) {
```